src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -486,6 +486,14 @@ public String getName() {
         return name;
     }
 
+    /**
+     * {@return the string representation of the ZIP file}
+     */
+    @Override
+    public String toString() {
+        return "" + name + Integer.toHexString(System.identityHashCode(this));
```

Comment:
This will give you a String like `foo/bar.zip174d20a`, I think you'll need some separator and "@" is often used with the hex representation of the identity hash. Using `name + "@" + Objects.toIdentityString(this);` is another option to consider.

Reply:
@justin-curtis-lu, for inspiration and consistency, you might want to have a look at this implementation: https://github.com/openjdk/jdk/blob/1d9688667e667dc710d64e52f1e918e047beaca3/src/java.base/share/classes/java/util/concurrent/CompletableFuture.java#L2605-L2628

Reply:
Generally, search codebase for `super.toString()` to get some idea as to how people incorporate superclass `toString()` into subclass one.

Reply:
Thanks Pavel and Alan. 

I updated the return value to be separated by '@'. I also updated the specification wording so that users will be cautious when using this value as @ecki suggested. I used phrasing similar to the second example in Effective Java Item 12. I expect the 'subject to change' will ensure people do not depend on it.